x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/history/HistoryStoreTests.java

```diff
@@ -105,14 +120,21 @@ public void testPut() throws Exception {
     }
 
     public void testIndexNameGeneration() {
-        String indexTemplateVersion = Integer.toString(INDEX_TEMPLATE_VERSION);
-        assertThat(getHistoryIndexNameForTime(Instant.ofEpochMilli((long) 0).atZone(ZoneOffset.UTC)),
-            equalTo(".watcher-history-"+ indexTemplateVersion +"-1970.01.01"));
-        assertThat(getHistoryIndexNameForTime(Instant.ofEpochMilli(100000000000L).atZone(ZoneOffset.UTC)),
+        String indexTemplateVersion;
+        if (randomBoolean()) {
```

Comment:
Why don't we about we extract this boolean to a parameter and always run both, rather than randomizing? This is a unit test, so it should be too expensive to do so, and that way the suite is guaranteed to catch breakage in a single test run.

Reply:
[done](https://github.com/elastic/elasticsearch/pull/56734/commits/123453ccfff325b6fa364944dd19a37bc7e789c2)